File: tests/integration/aliases/python/step2/scenario_extract_component_child.py
# Copyright 2026, Pulumi Corporation.  All rights reserved.

import pulumi


class FooResource(pulumi.ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:FooResource", name, None, opts)


class ComponentResource(pulumi.ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:ComponentResource", name, None, opts)


comp = ComponentResource("compExtractChild")

FooResource("childExtractChild", pulumi.ResourceOptions(
    aliases=[pulumi.Alias(parent=comp)]
))

File: tests/integration/aliases/go/step2/scenario_extract_component_child.go
// Copyright 2026, Pulumi Corporation.  All rights reserved.
//go:build !all
// +build !all

package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

type extractChildFooResource struct {
	pulumi.ResourceState
}

type extractChildFooComponent struct {
	pulumi.ResourceState
}

func newExtractChildFooResource(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*extractChildFooResource, error) {
	fooRes := &extractChildFooResource{}
	err := ctx.RegisterComponentResource("my:module:FooResource", name, fooRes, opts...)
	if err != nil {
		return nil, err
	}
	return fooRes, nil
}

func newExtractChildFooComponent(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*extractChildFooComponent, error) {
	fooComp := &extractChildFooComponent{}
	err := ctx.RegisterComponentResource("my:module:FooComponent", name, fooComp, opts...)
	if err != nil {
		return nil, err
	}
	return fooComp, nil
}

func scenarioExtractComponentChild(ctx *pulumi.Context) error {
	foo, err := newExtractChildFooComponent(ctx, "compExtractChild")
	if err != nil {
		return err
	}

	aliasOpt := pulumi.Aliases([]pulumi.Alias{{
		Parent: foo,
	}})
	_, err = newExtractChildFooResource(ctx, "childExtractChild", aliasOpt)
	return err
}
